File: ingester/internal/events/mint_event.go
package events

type MintEvent struct {
	BaseEvent
	Sender  string `json:"sender"`
	Amount0 string `json:"amount0"` // Token0 amount added to pool
	Amount1 string `json:"amount1"` // Token1 amount added to pool
}

func (e MintEvent) ToMap() map[string]interface{} {
	m := e.BaseEvent.ToMap()
	m["sender"] = e.Sender
	m["amount0"] = e.Amount0
	m["amount1"] = e.Amount1
	return m
}

File: aggregator/src/main/java/com/web3analytics/models/MintEvent.java
package com.web3analytics.models;

/**
 * Mint event from dex-liquidity-events topic.
 * Represents liquidity provision (token amounts added).
 */
public record MintEvent(
        String eventId,
        long blockNumber,
        long blockTimestamp,
        String transactionHash,
        int logIndex,
        String pairAddress,
        String token0,
        String token1,
        String token0Symbol,
        String token1Symbol,
        String sender,
        String amount0,
        String amount1,
        long eventTimestamp
) implements DexEvent {}
